neqo-transport/src/connection/mod.rs

```diff
@@ -1113,12 +1119,22 @@ impl Connection {
         delay
     }
 
+    /// Shorthand for [`Connection::process`] with no input `dgram`.
+    #[must_use = "Output of the process function must be handled"]
+    pub fn process_output(&mut self, now: Instant) -> Output {
+        self.process(None, now)
+    }
+
     /// Get output packets, as a result of receiving packets, or actions taken
     /// by the application.
     /// Returns datagrams to send, and how long to wait before calling again
     /// even if no incoming packets.
     #[must_use = "Output of the process_output function must be handled"]
-    pub fn process_output(&mut self, now: Instant) -> Output {
+    fn process_output_into_buffer<'a>(
+        &mut self,
+        now: Instant,
+        out: &'a mut Vec<u8>,
```

Comment:
Why `&mut Vec<u8>` and not `&mut [u8]`?  I don't think that -- as long as we are accepting someone else's memory -- we should be reallocating it.